src/constants/providers.ts

```diff
@@ -1,14 +1,49 @@
+import { deepCopy } from '@ethersproject/properties'
 // This is the only file which should instantiate new Providers.
 // eslint-disable-next-line @typescript-eslint/no-restricted-imports
 import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { isPlain } from '@reduxjs/toolkit'
 
 import { SupportedChainId } from './chains'
 import { RPC_URLS } from './networks'
 
 class AppJsonRpcProvider extends StaticJsonRpcProvider {
+  private _blockCache = new Map<string, Promise<any>>()
+  get blockCache() {
+    // If the blockCache has not yet been initialized this block, do so by
+    // setting a listener to clear it on the next block.
+    if (!this._blockCache.size) {
+      this.once('block', () => this._blockCache.clear())
```

Comment:
is the provider already subscribed per block? will this add a new subscription or use the existing one?

Reply:
Based on my read of the source code, the provider is only subscribed if there is a call (through `blockNumber`, `getBlockNumber()`, `on('block', ...)`, or `once('block', ...)`); so this is necessary to be notified when there is a new block, _and_ this will add a new subscription iff one does not already exist.

Caveat: I only skimmed the code - it's possible it will create an add'l subscription, but that doesn't seem bad for perf.